bin/propolis-server/src/lib/vm/request_queue.rs

```diff
@@ -459,48 +590,89 @@ mod test {
     }
 
     fn make_migrate_as_source_request() -> ExternalRequest {
-        ExternalRequest::MigrateAsSource {
+        ExternalRequest::State(StateChangeRequest::MigrateAsSource {
             migration_id: Uuid::new_v4(),
             websock: WebsocketConnection(None),
-        }
+        })
     }
 
     fn make_reconfigure_crucible_request() -> ExternalRequest {
         let (tx, _rx) = tokio::sync::oneshot::channel();
-        ExternalRequest::ReconfigureCrucibleVolume {
-            backend_id: SpecKey::Uuid(Uuid::new_v4()),
-            new_vcr_json: "".to_string(),
-            result_tx: tx,
+        ExternalRequest::Component(
+            ComponentChangeRequest::ReconfigureCrucibleVolume {
+                backend_id: SpecKey::Uuid(Uuid::new_v4()),
+                new_vcr_json: "".to_string(),
+                result_tx: tx,
+            },
+        )
+    }
+
+    impl ExternalRequest {
+        fn assert_start(&self) {
+            assert!(
+                matches!(self, Self::State(StateChangeRequest::Start)),
+                "expected start request, got {self:?}"
+            );
+        }
+
+        fn assert_stop(&self) {
+            assert!(
+                matches!(self, Self::State(StateChangeRequest::Stop)),
+                "expected stop request, got {self:?}"
+            );
+        }
+
+        fn assert_reboot(&self) {
+            assert!(
+                matches!(self, Self::State(StateChangeRequest::Reboot)),
+                "expected reboot request, got {self:?}"
+            );
+        }
+
+        fn assert_migrate_as_source(&self) {
+            assert!(
+                matches!(
+                    self,
+                    Self::State(StateChangeRequest::MigrateAsSource { .. })
+                ),
+                "expected migrate as source request, got {self:?}"
+            );
         }
     }
 
-    #[tokio::test]
-    async fn start_requests_become_idempotent_after_first_request() {
+    #[test]
+    fn start_requests_become_idempotent_after_first_request() {
```

Comment:
it occurs to me that if we wanted to be particularly fastidious, a lot of these tests _could_ probably be expressed as property tests, where we would generate a random sequence of requests and assert that the property holds across all possible requests once we have entered into a particular state...

Reply:
but, you know, what you've got seems good enough.

Reply:
(Un?)fortunately, I was just thinking the other day about how it'd be nice to exercise these muscles a bit. Added some property tests in 58cb957 and 81dde61.